newIDE/app/src/BehaviorsEditor/index.js

```diff
@@ -143,6 +170,118 @@ const BehaviorsEditor = (props: Props) => {
     if (props.onBehaviorsUpdated) props.onBehaviorsUpdated();
   };
 
+  const copyBehavior = (behaviorName: string) => {
+    const behavior = object.getBehavior(behaviorName);
+    Clipboard.set(BEHAVIORS_CLIPBOARD_KIND, [
+      {
+        name: behaviorName,
+        type: behavior.getTypeName(),
+        serializedBehavior: serializeToJSObject(behavior),
+      },
+    ]);
+  };
+
+  const copyAllBehaviors = () => {
+    Clipboard.set(
+      BEHAVIORS_CLIPBOARD_KIND,
+      mapVector(object.getAllBehaviorNames(), behaviorName => {
+        const behavior = object.getBehavior(behaviorName);
+        return {
+          name: behaviorName,
+          type: behavior.getTypeName(),
+          serializedBehavior: serializeToJSObject(behavior),
+        };
+      })
+    );
+  };
+
+  const newNamedBehavior: Array<{
+    name: string,
+    type: string,
+    serializedBehavior: string,
+  }> = [];
+  const existingNamedBehavior: Array<{
+    name: string,
+    type: string,
+    serializedBehavior: string,
+  }> = [];
+  const existingBehaviorFullNames: Array<string> = [];
+
+  const pasteBehaviors = async () => {
```

Comment:
I feel like the majority of the function could be extracted outside of the component. It would help the BehaviorsEditor component to be easier to re-render